packages/desktop-client/src/hooks/usePreviewTransactions.ts

```diff
@@ -1,34 +1,66 @@
-import { useMemo } from 'react';
+import { useState } from 'react';
 
 import {
   type ScheduleStatuses,
   useCachedSchedules,
 } from 'loot-core/client/data-hooks/schedules';
+import { send } from 'loot-core/platform/client/fetch';
+import { ungroupTransactions } from 'loot-core/shared/transactions';
 import { type ScheduleEntity } from 'loot-core/types/models';
 
-export function usePreviewTransactions() {
+import { type TransactionEntity } from '../../../loot-core/src/types/models/transaction.d';
+
+export function usePreviewTransactions(
```

Comment:
The data in this hook is now lazily loaded (since it has to make calls to async APIs), but the caller needs the data (or empty) non-lazily. I went with this approach so we can return the data via a `setState`. Does that make sense?